Testing/SystemTests/tests/qt/WorkbenchStartupTest.py

```diff
@@ -0,0 +1,56 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import os
+import subprocess
+import sys
+import systemtesting
+
+from mantid.kernel import ConfigService
+
+
+EXECUTABLE_SWITCHER = {"linux": ["launch_mantidworkbench.sh", "mantidworkbench"],
+                       "darwin": ["MantidWorkbench"],
+                       "win32": ["MantidWorkbench.exe"]}
+
+
+def get_mantid_executables_for_platform(platform):
+    workbench_executables = EXECUTABLE_SWITCHER.get(platform, None)
+    if workbench_executables is None:
+        raise RuntimeError("Unknown platform {0}.".format(platform))
+    return workbench_executables
+
+
+def get_mantid_executable_path(platform):
+    directory = ConfigService.getPropertiesDir().replace('\\', '/')
+    for executable in get_mantid_executables_for_platform(platform):
+        workbench_exe = os.path.join(directory, executable)
+        if os.path.exists(workbench_exe):
+            return workbench_exe
+    raise RuntimeError("Could not find path to {0}.".format(workbench_exe))
+
+
+class WorkbenchStartupTest(systemtesting.MantidSystemTest):
+    """
+    A system test for testing that Mantid Workbench opens ok for Linux, MacOS and Windows.
+    """
+    def __init__(self):
+        super(WorkbenchStartupTest, self).__init__()
+
+        self._executable = get_mantid_executable_path(sys.platform)
+
+    def runTest(self):
+        process = subprocess.Popen(self._executable, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        # Assert that mantid workbench is running
+        self.assertTrue(process.poll() is None)
+
+        out, _ = process.communicate()
+        self.assertTrue(b'ConfigService-[Information] This is Mantid' in out)
```

Comment:
This is not quite working as expected for me on Linux. I think on Linux this comes out on `stderr` and not `stdout`.

May I suggest that we use some of the existing workbench arguments to control it shutting down automatically and actually we can then put back the running a script test? This way we don't have to kill the process and we can test a clean start/shutdown. It is possible to run workbench like:
```
executable_path --execute hello.py --quit
```
where `hello.py` is just a normal python script. Workbench will run the script and then quit afterwards with an exit code. Note that there is a slight bug in this at the moment so even more reason to get it tested! The line in [asynchronous.py#L20](https://github.com/mantidproject/mantid/blob/master/qt/python/mantidqt/utils/asynchronous.py#L20) should be `class TaskExitCode(IntEnum):` so that `int()` works as expected.

What do you think? We can just write the test script to a temporary directory as part of the test, do something like `print("Hello")` and check we get that back?